rest_framework-stubs/reverse.pyi

```diff
@@ -1,21 +1,21 @@
 from typing import Any, Callable, Mapping, Optional, Sequence
 
-from rest_framework.request import Request
+from django.http import HttpRequest
```

Comment:
why this change? It does not appear to be correct lookint at the code.

Reply:
show how it can break the code please, Request is subclass of the HttpRequest

Reply:
Thats not a valid reason. It can break the code because request is not httrequest and the DRF deals with request and not http request.

Reply:
show how this code will break if I pass HttpRequest into these functions

Reply:
For me it looks like `HttpRequest` can be used: https://github.com/encode/django-rest-framework/blob/master/rest_framework/reverse.py

Reply:
@kalekseev - i asked you why you introduced this change, your answer "show how it break...". This is a poor answer. if there is no good reason to change the typing here, then dont please.

Reply:
@Goldziher this is a pretty valid question. It  is about variance, since `Request` is a sub-type of `HttpRequest`.

Btw, we encourage people to contribute to our repos. Not discourage them to. 
So, any valid contributions: bugfixes, improvements, docs, etc are welcome 🙂

Reply:
ofc, but is this an improvement? If there is a viable reason for doing this, e.g. typing this with Request causes an issue etc. than its fine, but if its just because of reasons then I am against this. 

In fact one of the most persistent issues I have with version 1.2 is that HttpRequest is used in multiple places where the DRF in fact uses Request, this causes MyPy to complain and I have to type ignore valid code.

The DRF uses Request, not HttpRequest, and Request has properties that are not available on HttpRequest. So again - is there a valid reason for making this change here?

Reply:
one more thing @kalekseev  - i do encourage and appreciate your efforts (and knowledge)! These review comments are just code review - not person in any way.

Reply:
@Goldziher relax man, I'm too old to care about someone words on the internet. Unfortunately code review is a hard work actually, I do them a lot and know that to make an effective review I need to understand the problem, load full context into my head and make research if I don't know some topic. So what I see from your comments and code is that you try to give a very specific type for all the things but actually in type theory this works another way, you want to give the least specific types for function arguments because the more types function accepts the more valuable it is. For example if function accepts list and just iterate on it, you could use argument type List but tomorrow you will find that you need same function for Set so it's better to make your function argument `Iterable`. In this particular case `reverse` function can work with `HttpResponse` without problem, it doesn't use any `Response` specific attributes or anything else so why would you want to constraint user to use `reverse` only with `Response` while function happily accepts HttpResponse and all its subtypes?

Reply:
Fair enough, thanks for you explaining your PoV